src/analysis/retail/deathknight/unholy/modules/talents/SummonGargoyleBuffs.tsx

```diff
@@ -0,0 +1,128 @@
+import { t } from '@lingui/macro';
+import SPELLS from 'common/SPELLS';
+import TALENTS from 'common/TALENTS/deathknight';
+import { SpellLink } from 'interface';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { CastEvent } from 'parser/core/Events';
+import { When } from 'parser/core/ParseResults';
+import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+import SpellUsable from 'parser/shared/modules/SpellUsable';
+
+class SummonGargoyleBuffs extends Analyzer {
+  static dependencies = {
+    spellUsable: SpellUsable,
+  };
+
+  protected spellUsable!: SpellUsable;
+
+  totalGargoyleCasts = 0; // Number of fully completed gargoyles used
+  currentGargoyleRunicPower = 0; // Keeps track of Runic Power used on current active Gargoyle
+  totalGargoyleRunicPower = 0; // Total Runic Power spend on all fully completed Gargoyles
+  gargoyleActive = false; // Boolean to keep track of Gargoyle being active or not
+
+  constructor(options: Options) {
+    super(options);
+
+    this.active = this.selectedCombatant.hasTalent(TALENTS.SUMMON_GARGOYLE_TALENT);
+    if (!this.active) {
+      return;
+    }
+
+    this.addEventListener(
+      Events.cast
+        .by(SELECTED_PLAYER)
+        .spell([SPELLS.DEATH_COIL, SPELLS.EPIDEMIC, SPELLS.DEATH_STRIKE_HEAL]),
+      this.onBuffCast,
+    );
+
+    this.addEventListener(
+      Events.cast
+        .by(SELECTED_PLAYER)
+        .spell([SPELLS.SUMMON_GARGOYLE, SPELLS.DARK_ARBITER_TALENT_GLYPH]),
+      this.onGargCast,
+    );
+  }
+
+  onBuffCast(event: CastEvent) {
+    console.log(event.ability.name);
+    if (
+      this.gargoyleActive === true &&
+      this.spellUsable.cooldownRemaining(SPELLS.SUMMON_GARGOYLE.id, event.timestamp) < 155000
+    ) {
```

Comment:
this is probably safer to do by tracking the `summon` event (https://www.warcraftlogs.com/reports/2mATn7bL1RdKjvD6#fight=11&source=54&type=summary&view=events&pins=2%24Off%24%23244F4B%24expression%24type%3D%22summon%22) and then using `timestamp + 25000` as the cutoff.

That way, if there are CDR things in the future this won't quietly break.

If you want this to also be available to other modules, you can also add a normalizer that adds a death event where it *should* exist.

Reply:
I changed it to look for gargoyle cast and then set a time in the future when it ends, using that to figure out if it's done or not.